src/controllers/import.js

```diff
@@ -125,6 +127,63 @@ function ImportController(context) {
     }
   }
 
+  /**
+   * Extract the domain from a URL.
+   * @param inputUrl
+   * @return domain
+   */
+  function getDomain(inputUrl) {
+    const parsedUrl = new URL(inputUrl);
+    const parsedDomain = psl.parse(parsedUrl.hostname);
+    return parsedDomain.domain; // Extracts the full domain (e.g., example.co.uk)
+  }
+
+  /**
+   * Check if the URLs in urlList belong to any of the base domains.
+   * @param urlList
+   * @param baseDomainList
+   * @return {true} if all URLs belong to an allowed base domain
+   * @throws {ErrorWithStatusCode} if any URL does not belong to an allowed base domain
+   */
+  function isUrlInBaseDomains(urlList, baseDomainList) {
+    return urlList.every((inputUrl) => {
+      const urlDomain = getDomain(inputUrl);
+      const belongsToBaseDomain = baseDomainList.some((baseDomain) => urlDomain === baseDomain);
+      if (!belongsToBaseDomain) {
+        throw new ErrorWithStatusCode(`Invalid request: URL: ${inputUrl} not allowed`, STATUS_BAD_REQUEST);
+      }
```

Comment:
Let's collect all the invalid URLs then throw the error if the list has a length > 0.  That way the user isn't going back and forth potentially.  We should be able to tell them the potential list of invalid URLs they provided.